colexecdisk: extract disk-backed operators into a new package

yuzefovich opened the pull request:
**colexecdisk: extract disk-backed operators into a new package**

This allows us to break dependency of `colexec` on `colexecjoin` which
speeds up the build a bit.

Addresses: #79357.

Release note: None

**execagg: extract out some aggregate helpers into a separate package**

This allows us to break dependency of `colexecagg` on `execinfra`.

Release note: None

yuzefovich committed:
colexecdisk: extract disk-backed operators into a new package

This allows us to break dependency of `colexec` on `colexecjoin` which
speeds up the build a bit.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80810)
<!-- Reviewable:end -->

yuzefovich committed:
execagg: extract out some aggregate helpers into a separate package

This allows us to break dependency of `colexecagg` on `execinfra`.

Release note: None

msirek reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80810#-:-N14X0FV18h5a17bq7X6:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @cucaroach)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+

cucaroach reviewed (approved):
> Reviewed 28 of 28 files at r1, 19 of 19 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80810)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->
